Bio/Entrez/__init__.py

```diff
@@ -187,13 +187,14 @@ def efetch(db, **keywords):
     except KeyError:
         pass
     else:
-        try:
-            ids = ",".join(ids)
-        except TypeError as e:
-            if isinstance(ids, int):
-                ids = str(ids)
-            else:
-                raise e
+        if not isinstance(ids, str):
+            try:
+                ids = ",".join(ids)
+            except TypeError as e:
+                if isinstance(ids, int):
+                    ids = str(ids)
+                else:
+                    raise e
```

Comment:
`raise e` will give a double traceback

Reply:
Actually, the `raise e` was there before, but I can try to get this issue solved.

Reply:
Sorry, I was wrong, the `raise e` won't give a double traceback.

Reply:
But it could be reduced to just ``raise`` though, right?